src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -213,6 +213,24 @@ public ConfidentialClientApplicationBuilder WithAzureRegion(string azureRegion =
             return this;
         }
 
+
+        /// <summary>
+        /// When set to <c>true</c>, MSAL will lock cache access at the <see cref="ConfidentialClientApplication"/> level, i.e.
+        /// the block of code between BeforeAccessAsync and AfterAccessAsync callbacks will be syncronized. 
+        /// Apps can set this flag to <c>false</c> to enable an optimistic cache locking strategy, which may result in better performance, especially 
+        /// when ConfidentialClientApplication objects are reused.
+        /// </summary>
+        /// <remarks>
+        /// True by default, but subject to change.
+        /// Not recommended for apps that call RemoveAsync
+        /// </remarks>
+        public ConfidentialClientApplicationBuilder WithCacheSynchronization(bool enableCacheSyncronization)
```

Comment:
Would we want to have it in ConfidentialClientApplicationOptions as well?

Reply:
@bgavrilMS and I discussed that on Friday. He could do both.

Reply:
yes, that was my understanding.

Reply:
Will do